### PR TITLE
Made predictions device agnostic

### PR DESCRIPTION
**Pull request by mgupta27**
Made predictions device agnostic so that the code runs on machines with and without CUDA. After setting up the code on my laptop I ran into this error so I fixed it so no one should run into these errors now.

### CODE REVIEW

File: modules/detect_target/detect_target.py [hunk 1]
```diff
@@ -22,6 +23,7 @@ def __init__(self, model_path: str, save_name: str=""):
         self.__model = ultralytics.YOLO(model_path)
         self.__counter = 0
         self.__filename_prefix = save_name + "_" + str(int(time.time())) + "_"
+        self.__device = 0 if torch.cuda.is_available() else "cpu"
```
Comment (Xierumeng) [inline modules/detect_target/detect_target.py:26]: Nit: Move to the line after `self.__model`

Personally, I don't like this format of conditional assignment but that's just my preference, fine to leave as is.

**Comment by Xierumeng** [comment]
Please rebase on or merge from latest main to resolve conflicts.

**Comment by Xierumeng** [comment]
Next time also please use the WARG repository directly rather than a personal fork.

File: modules/detect_target/detect_target.py (hunk 1 above)
Comment (mgupta27) [reply modules/detect_target/detect_target.py:24]: Ok in that case we can leave it. Maybe we should just make a note of it in the documentation so people know.

**Review by Xierumeng — APPROVED** [review]

**Pull request closed without merging**